plugin-cloud-storage: Can't resolve 'fs' error with cloud-storage-plugin s3 adapter

### Link to reproduction

https://github.com/haliacmon/payload-reproduction

### Describe the Bug

Getting Module not found: Error: Can't resolve 'fs' error with plugin-cloud-storage and s3-adapter

```
ERROR in ./node_modules/.pnpm/@payloadcms+plugin-cloud-storage@1.1.1_@aws-sdk+client-s3@3.454.0_@aws-sdk+lib-storage@3.454.0_payload@2.1.1/node_modules/@payloadcms/plugin-cloud-storage/dist/adapters/s3/handleUpload.js 140:27-40
Module not found: Error: Can't resolve 'fs' in '...\node_modules\.pnpm\@payloadcms+plugin-cloud-storage@1.1.1_@aws-sdk+client-s3@3.454.0_@aws-sdk+lib-storage@3.454.0_payload@2.1.1\node_modules\@payloadcms\plugin-cloud-storage\dist\adapters\s3'
```

### To Reproduce

1- npx create-payload-app@latest reproduce-error-s3 -t blank --use-pnpm
using postgress
2- pnpm add @payloadcms/plugin-cloud-storage
3- pnpm add @aws-sdk/client-s3 @aws-sdk/lib-storage aws-crt
4- optional. add s3 related variables to .env. Not necessary to reproduce the error.
5- adding build config for plugin-cloud-storage & s3-adapter by instructions. (error happens after step 5)
https://github.com/payloadcms/payload/tree/c10db332cdeb3f702d378b0649be20eb789c13e1/packages/plugin-cloud-storage#s3-adapter

```javascript
//payload.config.ts
...
import { cloudStorage } from '@payloadcms/plugin-cloud-storage'
import { s3Adapter } from '@payloadcms/plugin-cloud-storage/s3'

const adapter = s3Adapter({
  config: {
    credentials: {
      accessKeyId: process.env.S3_ACCESS_KEY_ID,
      secretAccessKey: process.env.S3_SECRET_ACCESS_KEY,
    },
    region: process.env.S3_REGION,
  },
  bucket: process.env.S3_BUCKET,
})
...

plugins: [
...
    cloudStorage({
      collections: {
        "my-collection-slug": {
          adapter: adapter,
        },
      },
    }),
  ],
...
```


### Payload Version

2.1.1

### Adapters and Plugins

db-postgres, plugin-cloud-storage, S3 Adapter

--- a/packages/plugin-cloud-storage/package.json
+++ b/packages/plugin-cloud-storage/package.json
@@ -16,8 +16,8 @@
   "peerDependencies": {
     "@aws-sdk/client-s3": "^3.142.0",
     "@aws-sdk/lib-storage": "^3.267.0",
-    "@azure/storage-blob": "^12.11.0",
     "@azure/abort-controller": "^1.0.0",
+    "@azure/storage-blob": "^12.11.0",
     "@google-cloud/storage": "^6.4.1",
     "payload": "^1.7.2 || ^2.0.0"
   },
@@ -49,6 +49,7 @@
     "@azure/storage-blob": "^12.11.0",
     "@google-cloud/storage": "^6.4.1",
     "@types/express": "^4.17.9",
+    "@types/find-node-modules": "^2.1.2",
     "cross-env": "^7.0.3",
     "dotenv": "^8.2.0",
     "nodemon": "^2.0.6",
@@ -58,6 +59,7 @@
     "webpack": "^5.78.0"
   },
   "dependencies": {
+    "find-node-modules": "^2.1.3",
     "range-parser": "^1.2.1"
   }
 }

--- a/packages/plugin-cloud-storage/src/adapters/s3/webpack.ts
+++ b/packages/plugin-cloud-storage/src/adapters/s3/webpack.ts
@@ -1,15 +1,37 @@
 import type { Configuration as WebpackConfig } from 'webpack'
 
+import findNodeModules from 'find-node-modules'
+import fs from 'fs'
 import path from 'path'
 
+const packageName = '@payloadcms/plugin-cloud-storage'
+
+const nodeModulesPaths = findNodeModules({ cwd: __dirname, relative: false })
+
 export const extendWebpackConfig = (existingWebpackConfig: WebpackConfig): WebpackConfig => {
+  let nodeModulesPath = nodeModulesPaths.find((p) => {
+    const guess = path.resolve(p, `${packageName}/dist`)
+
+    if (fs.existsSync(guess)) {
+      return true
+    }
+    return false
+  })
+
+  if (!nodeModulesPath) {
+    nodeModulesPath = process.cwd()
+  }
+
   const newConfig: WebpackConfig = {
     ...existingWebpackConfig,
     resolve: {
       ...(existingWebpackConfig.resolve || {}),
       alias: {
         ...(existingWebpackConfig.resolve?.alias ? existingWebpackConfig.resolve.alias : {}),
-        '@payloadcms/plugin-cloud-storage/s3': path.resolve(__dirname, './mock.js'),
+        '@payloadcms/plugin-cloud-storage/s3$': path.resolve(
+          nodeModulesPath,
+          `./${packageName}/dist/adapters/s3/mock.js`,
+        ),
       },
       fallback: {
         ...(existingWebpackConfig.resolve?.fallback ? existingWebpackConfig.resolve.fallback : {}),

--- a/packages/plugin-cloud-storage/src/webpack.ts
+++ b/packages/plugin-cloud-storage/src/webpack.ts
@@ -29,7 +29,7 @@ export const extendWebpackConfig =
       },
     }
 
-    return Object.entries(options.collections).reduce(
+    const modifiedConfig = Object.entries(options.collections).reduce(
       (resultingWebpackConfig, [slug, collectionOptions]) => {
         const matchedCollection = config.collections?.find((coll) => coll.slug === slug)
 
@@ -47,4 +47,6 @@ export const extendWebpackConfig =
       },
       newConfig,
     )
+
+    return modifiedConfig
   }

